frontend/src/__tests__/cypress/cypress/tests/mocked/connectionTypes/connectionTypes.cy.ts

```diff
@@ -0,0 +1,102 @@
+import { pageNotfound } from '~/__tests__/cypress/cypress/pages/pageNotFound';
+import {
+  asProductAdminUser,
+  asProjectAdminUser,
+} from '~/__tests__/cypress/cypress/utils/mockUsers';
+import { connectionTypesPage } from '~/__tests__/cypress/cypress/pages/connectionTypes';
+import { mockDashboardConfig } from '~/__mocks__';
+import { mockConnectionTypeConfigMap } from '~/__mocks__/mockConnectionType';
+
+it('Connection types should not be available for non product admins', () => {
+  asProjectAdminUser();
+  cy.visitWithLogin('/connectionTypes');
+  pageNotfound.findPage().should('exist');
+  connectionTypesPage.findNavItem().should('not.exist');
+});
+
+it('Connection types should be hidden by feature flag', () => {
+  asProductAdminUser();
+
+  cy.visitWithLogin('/connectionTypes');
+  connectionTypesPage.shouldReturnNotFound();
+
+  cy.interceptOdh(
+    'GET /api/config',
+    mockDashboardConfig({
+      disableConnectionTypes: false,
+    }),
+  );
+
+  connectionTypesPage.visit();
+});
+
+describe('Connection types', () => {
+  beforeEach(() => {
+    asProductAdminUser();
+
+    cy.interceptOdh(
+      'GET /api/config',
+      mockDashboardConfig({
+        disableConnectionTypes: false,
+      }),
+    );
+    cy.interceptOdh('GET /api/connection-types', [
+      mockConnectionTypeConfigMap({}),
+      mockConnectionTypeConfigMap({
+        name: 'no-display-name',
+        displayName: '',
+        description: 'description 2',
+        enabled: false,
+        preInstalled: true,
+      }),
+      mockConnectionTypeConfigMap({
+        name: 'test-2',
+        displayName: 'Test display name',
+        description: 'Test description',
+      }),
+    ]);
+  });
+
+  it('should show the connections type table', () => {
+    connectionTypesPage.visit();
+    connectionTypesPage.shouldHaveConnectionTypes();
+  });
```

Comment:
No need for fine granular tests. This will just increase our test time.
This is a redundant test with the other test that inspects columns.